src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1466,8 +1474,9 @@ int XrdHttpReq::ProcessHTTPReq() {
 
           TRACEI(REQ, "Writing " << bytes_to_read);
           if (!prot->Bridge->Run((char *) &xrdreq, prot->myBuffStart, bytes_to_read)) {
-            prot->SendSimpleResp(404, NULL, NULL, (char *) "Could not run write request.", 0, false);
-            return -1;
+            TRACEI(REQ, " Failed to run write request on the bridge.");
```

Comment:
Maybe you can put the error message in a string allowing you to not repeat the same string in the two calls?

Reply:
Done in latest commit.